effect/shared/src/main/scala/scalaz/effect/IO.scala

```diff
@@ -164,7 +165,10 @@ sealed abstract class IO[E, A] { self =>
    * Widens the error type to any supertype. While `leftMap` suffices for this
    * purpose, this method is significantly faster for this purpose.
    */
-  final def widen[E2 >: E]: IO[E2, A] = self.asInstanceOf[IO[E2, A]]
+  final def widen[E2](implicit ev: E <~< E2): IO[E2, A] = {
+    val _ = ev
+    self.asInstanceOf[IO[E2, A]]
```

Comment:
Ideally we should have `IsCovariant` instances for `IO[?, A]` and `IO[A, ?]` (or perhaps `widenLeft` on `Bifunctor`). But this is out of scope for this PR.